### PR TITLE
Pass time and period to read() and write()

### PR DESCRIPTION
**Pull request by bmagyar**
Counterpart of https://github.com/ros-controls/ros2_control/pull/715

### CODE REVIEW

**Review by destogl — APPROVED** [review]
Great!

File: ros2_control_demo_hardware/src/diffbot_system.cpp [hunk 1]
```diff
@@ -179,10 +179,11 @@ hardware_interface::CallbackReturn DiffBotSystemHardware::on_deactivate(
   return hardware_interface::CallbackReturn::SUCCESS;
 }
 
-hardware_interface::return_type DiffBotSystemHardware::read()
+hardware_interface::return_type DiffBotSystemHardware::read(
+  const rclcpp::Time & time, const rclcpp::Duration & period)
 {
-  current_timestamp = clock_.now();
-  rclcpp::Duration dt = current_timestamp - last_timestamp_;  // Control period
+  current_timestamp = time;
```
Comment (destogl) [inline ros2_control_demo_hardware/src/diffbot_system.cpp:185]: Can you please also clean initialization of `clock_` and members `last_timestamp_` from the files. I think they are not needed.

**Review by destogl — CHANGES_REQUESTED** [review]
I would still like to get a few further adjustments in if possible.

File: ros2_control_demo_hardware/src/diffbot_system.cpp (hunk 1 above)
Comment (bmagyar) [reply ros2_control_demo_hardware/src/diffbot_system.cpp]: done

**Review by destogl — APPROVED** [review]
Thanks!

**Review by anfemosa — APPROVED** [review]
Great!

**Pull request merged**